src/transformers/models/ernie_m/tokenization_ernie_m.py

```diff
@@ -0,0 +1,354 @@
+# and Albert directory.(https://github.com/huggingface/transformers/blob/main/src/transformers/models/albert/tokenization_albert.py)
+import io
+import unicodedata
+from typing import List, Optional
+
+import sentencepiece as spm
+
+from ...tokenization_utils import PreTrainedTokenizer
+
+
+__all__ = ["ErnieMTokenizer"]
+
+SPIECE_UNDERLINE = "▁"
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt", "sentencepiece_model_ckpt": "sentencepiece.bpe.model"}
+
+
+class ErnieMTokenizer(PreTrainedTokenizer):
+    r"""
+    Constructs a ErnieM tokenizer. It uses the `sentencepiece` tools to cut the words to sub-words.
+
+    Args:
+        vocab_file (`str`):
+            The file path of the vocabulary.
+        sentencepiece_model_file (`str`):
+            The file path of sentencepiece model.
+        do_lower_case (`str`, *optional*, defaults to True):
+            Whether or not to lowercase the input when tokenizing.
+        unk_token (`str`, *optional*, defaults to "[UNK]"):
+            A special token representing the `unknown (out-of-vocabulary)` token. An unknown token is set to be
+            `unk_token` inorder to be converted to an ID.
+        sep_token (`str`, *optional*, defaults to "[SEP]"):
+            A special token separating two different sentences in the same input.
+        pad_token (`str`, *optional*, defaults to "[PAD]"):
+            A special token used to make arrays of tokens the same size for batching purposes.
+        cls_token (`str`, *optional*, defaults to "[CLS]"):
+            A special token used for sequence classification. It is the last token of the sequence when built with
+            special tokens.
+        mask_token (`str`, *optional*, defaults to "[MASK]"):
+            A special token representing a masked token. This is the token used in the masked language modeling task
+            which the model tries to predict the original unmasked ones.
+    """
+    resource_files_names = {
+        "sentencepiece_model_file": "sentencepiece.bpe.model",
+        "vocab_file": "vocab.txt",
+    }  # for save_pretrained
+    pretrained_resource_files_map = {
+        "vocab_file": {
+            "ernie-m-base": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/vocab.txt",
+            "ernie-m-large": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/vocab.txt",
+        },
+        "sentencepiece_model_file": {
+            "ernie-m-base": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/sentencepiece.bpe.model",
+            "ernie-m-large": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/sentencepiece.bpe.model",
+        },
+    }
+    pretrained_init_configuration = {
+        "ernie-m-base": {"do_lower_case": False},
+        "ernie-m-large": {"do_lower_case": False},
+        "uie-m-base": {"do_lower_case": False},
+        "uie-m-large": {"do_lower_case": False},
+    }
```

Comment:
Those should all be defined in constant before, to be consistent with other tokenizers.